src/python/pants/backend/python/util_rules/pex_from_targets_test.py

```diff
@@ -427,19 +438,31 @@ def test_issue_12222(rule_runner: RuleRunner) -> None:
             ),
         }
     )
+
+    if enable_resolves:
+        options = [
+            "--python-enable-resolves",
+            # NB: Because this is a synthetic lockfile without a header.
+            "--python-invalid-lockfile-behavior=ignore",
+        ]
+    else:
+        options = [
+            f"--python-requirement-constraints={lockfile}",
+            "--python-resolve-all-constraints",
+        ]
+    rule_runner.set_options(options, env_inherit={"PATH"})
+
     request = PexFromTargetsRequest(
         [Address("", target_name="lib")],
         output_filename="demo.pex",
         internal_only=False,
         platforms=PexPlatforms(["some-platform-x86_64"]),
     )
-    rule_runner.set_options(
-        [
-            "--python-requirement-constraints=constraints.txt",
-            "--python-resolve-all-constraints",
-        ],
-        env_inherit={"PATH"},
-    )
     result = rule_runner.request(PexRequest, [request])
 
-    assert result.requirements == PexRequirements(["foo"], constraints_strings=constraints)
+    # TODO: Only a `resolve_constraints` file will be used as constraints for a cross-platform
+    # build currently, because we don't parse a manually generated (or poetry) lockfile to find
+    # constraints.
```

Comment:
@Eric-Arellano : This is a new issue in `2.10.x`, which we might want to pick a fix for... maybe even as part of this PR.

Reply:
Oh, that is tricky. So essentially we would have to use the lock file but strip the hashes from it, right? And that is only relevant when we are not using A pex native lock file?

Reply:
Yea. The existing ["parse the constraints file into requirements strings" code](https://github.com/pantsbuild/pants/blob/511691657d00827e5e44ab5118688b99c221d744/src/python/pants/backend/python/util_rules/pex_from_targets.py#L546-L632) _should_ make that easyish though...?

Reply:
Maybe this code:

https://github.com/pantsbuild/pants/blob/511691657d00827e5e44ab5118688b99c221d744/src/python/pants/backend/python/target_types.py#L1021-L1039

Reply:
@stuhood correct. It does the right thing and errors if it can't. This is at --lock resolve time though, which is a bit late. https://github.com/pantsbuild/pex/issues/1595 tracks failing the lock at creation time if `--platform` is in the mix and the results can't be resolved for each platform due to sdists, etc.

Reply:
Ok: I've added one more commit on top to actually parse/apply constraints from non-PEX lockfiles.